File: code_samples/php/utilities_string_manipulation.php
<?php
/**
 * Utilities: String Manipulation
 * AI/ML Training Sample
 */

class StringManipulation {
    private $data;
    
    public function __construct() {
        $this->data = "";
    }
    
    public function process($input) {
        $this->data = $input;
    }
    
    public function getData() {
        return $this->data;
    }
    
    public function validate() {
        return !empty($this->data);
    }
}

// Example usage
$instance = new StringManipulation();
$instance->process("example");
echo "Data: " . $instance->getData() . "\n";
echo "Valid: " . ($instance->validate() ? "true" : "false") . "\n";
?>

File: code_samples/go/utilities_string_manipulation.go
package main

import (
    "fmt"
)

// Utilities: String Manipulation
// AI/ML Training Sample

type StringManipulation struct {
    Data string
}

func NewStringManipulation() *StringManipulation {
    return &StringManipulation{
        Data: "",
    }
}

func (s *StringManipulation) Process(input string) {
    s.Data = input
}

func (s *StringManipulation) Validate() bool {
    return len(s.Data) > 0
}

func (s *StringManipulation) GetData() string {
    return s.Data
}

func main() {
    instance := NewStringManipulation()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
